### PR TITLE
Use enums for constants

### PR DESCRIPTION
**Pull request by ArchdukeTim**

### CODE REVIEW

File: hal-base/hal/constants.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import enum
```
Comment (ArchdukeTim) [inline hal-base/hal/constants.py]: Not sure if we wanna use enums for this class. It doesn't *hurt*, but I'm not sure if it helps now or will later

File: hal-base/hal/constants.py [hunk 2]
```diff
@@ -7,12 +8,12 @@
...
 
 
-class RuntimeType:
+class RuntimeType(enum.IntEnum):
     Athena = 0
     Mock = 1
 
 
-class UsageReporting:
+class UsageReporting(enum.IntEnum):
```
Comment (auscompgeek) [inline hal-base/hal/constants.py]: Given that this has a lot of duplicates with different semantics, I don't think this makes sense as a single enum.

Comment (ArchdukeTim) [reply hal-base/hal/constants.py]: We probably don't need to. I don't think anything is type hinted as UsageReporting

**Comment by virtuald** [comment]
Looks like it needs formatting.

File: hal-base/hal/constants.py (hunk 2 above)
Comment (virtuald) [reply hal-base/hal/constants.py]: Yeah, this just causes problems:

```
>>> hal.UsageReporting.kLanguage_LabVIEW
<UsageReporting.kResourceType_Module: 1>
```

**Pull request merged**